### ISSUE TITLE
Info extension -  Incorrect type name when using Types::Array

### ISSUE THREAD

**Issue by aglushkov** [dry-rb/dry-schema#394]
## Describe the bug

When using Info extension, info shows `type: :string` for Types::Array(:string)

## To Reproduce

```ruby
schema = Dry::Schema.Params do
  required(:opt1).filled(Types::Array)
  required(:opt2).filled(Types::Array(:string))
  required(:opt3).filled(Types::Array(:integer))
  required(:opt4).filled(Types::Array(:bool))
end   

schema.info

# => {:keys=>
  {:opt1=>{:required=>true, :type=>"array"},
   :opt2=>{:required=>true, :type=>"string"},
   :opt3=>{:required=>true, :type=>"integer"},
   :opt4=>{:required=>true, :type=>"bool"}}}
```

## Expected behavior

Type should be `array` always, or some more smart name like `array(string)`

## My environment

- dry-schema 1.8.0
- Affects my production application: **NO**
- Ruby version: 3.0.0
- OS: Ubuntu 18

**Comment by CandyFet** [comment]
Hello! I would like to help with fixing this bug. Which name should type have?

**Comment by solnic** [comment]
I'd say it should produce `type: "array", member: "string"` in case of `Array(:string)` etc.

**Comment by solnic** [comment]
Fixed via #397
